src/snowflake/connector/config_parser/__init__.py

```diff
@@ -0,0 +1,33 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+from __future__ import annotations
```

Comment:
It's enforced by [our pre-commit config](https://github.com/snowflakedb/snowflake-connector-python/blob/d6a47498eefee8c125ad37cecc2361f6ef66da45/.pre-commit-config.yaml#L55).
I added this because I wanted us to be ready for what seems to be the future when it comes.

Reply:
I figured it was something like that.  Okay, I'll put a pin in this style nit for now :D